src/hive_mind_bridge/test/SocketToMoveByIntegrationTest.cpp

```diff
@@ -116,26 +116,24 @@ int main(int argc, char** argv) {
              "\tPayload: %d",
              statusReturnFunctionName.c_str(), arg0);
 
-    // Send moveBy request
-    serializer.serializeToStream(moveByMessageDTO);
-    std::this_thread::sleep_for(std::chrono::milliseconds(1000));
-
-    // Listen for a response
-    MessageDTO message;
-    deserializer.deserializeFromStream(message);
-    ResponseDTO response = std::get<ResponseDTO>(message.getMessage());
-    UserCallResponseDTO userCallResponse = std::get<UserCallResponseDTO>(response.getResponse());
-    FunctionCallResponseDTO functionCallResponse =
-        std::get<FunctionCallResponseDTO>(userCallResponse.getResponse());
-    GenericResponseDTO genericResponse = functionCallResponse.getResponse();
-    GenericResponseStatusDTO status = genericResponse.getStatus();
-    std::string details = genericResponse.getDetails();
-    ROS_INFO("RESPONSE FROM HOST (moveBy): \n"
-             "\tResponse status: %d\n"
-             "\tDetails: %s",
-             status, details.c_str());
-
-    // Listen for a return message
-    MessageDTO returnMessage;
-    deserializer.deserializeFromStream(returnMessage);
+    for (int i = 0; i < 5; i++) {
+        // Send moveBy request
+        serializer.serializeToStream(moveByMessageDTO);
+
+        // Listen for a response
+        MessageDTO message;
+        deserializer.deserializeFromStream(message);
+        ResponseDTO response = std::get<ResponseDTO>(message.getMessage());
+        UserCallResponseDTO userCallResponse =
+            std::get<UserCallResponseDTO>(response.getResponse());
+        FunctionCallResponseDTO functionCallResponse =
+            std::get<FunctionCallResponseDTO>(userCallResponse.getResponse());
```

Comment:
Just  so you know std::get throws an exception if the type doesn't match, I see that it's a test so I guess it can be desired sometimes. thoug idk if it is. You can use std::get_if and check if it's null.

If not you can use std::visit though it's a bit more tricky to use

You can see how it's done here with std::get_if
https://github.com/SwarmUS/Propolis/blob/4eae4a4a8bd94e76a414b18ad64490086e840687/src/pheromones/src/MessageDTO.cpp#L90

Reply:
I wrote that code, I think Étienne did just a few modifications so I'll answer. This is a test main. We want it to throw if there is a problem.